backend/src/apiserver/storage/run_store.go

```diff
@@ -350,6 +353,25 @@ func (s *RunStore) CreateRun(r *model.RunDetail) (*model.RunDetail, error) {
 }
 
 func (s *RunStore) UpdateRun(runID string, condition string, workflowRuntimeManifest string) (err error) {
+	tx, err := s.db.DB.Begin()
+	if err != nil {
+		return util.NewInternalServerError(err, "transaction creation failed")
+	}
+
+	row := tx.QueryRow("SELECT WorkflowRuntimeManifest FROM run_details WHERE UUID = ? FOR UPDATE", runID)
+	var storedManifest string
+	if err := row.Scan(&storedManifest); err != nil {
+		tx.Rollback()
+		return util.NewResourceNotFoundError("Run", runID)
+	}
+
+	if s.metadataStore != nil {
```

Comment:
Could we encapsulate the metadata store operations in separate functions? 

Should the "if s.metadaStore != nil" happen before the first query?